src/keybindings.c

```diff
@@ -194,12 +194,68 @@ GeanyKeyBinding *keybindings_set_item(GeanyKeyGroup *group, gsize key_id,
 	kb->default_key = key;
 	kb->default_mods = mod;
 	kb->callback = callback;
+	kb->cb_func = NULL;
+	kb->cb_data = NULL;
 	kb->menu_item = menu_item;
 	kb->id = key_id;
 	return kb;
 }
 
 
+/** Creates a new keybinding using a GeanyKeyBindingFunc and attaches it to a keybinding group
+ *
+ * If given the callback should return @c TRUE if the keybinding was handled, otherwise @c FALSE
+ * to allow other callbacks to be run. This allows for multiplexing keybindings on the same keys,
+ * depending on the focused widget (or context). If the callback is NULL the group's callback will
+ * be invoked, but the same rule applies.
+ *
+ * @param group Group.
+ * @param key_id Keybinding index for the group.
+ * @param key (Lower case) default key, e.g. @c GDK_j, but usually 0 for unset.
+ * @param mod Default modifier, e.g. @c GDK_CONTROL_MASK, but usually 0 for unset.
+ * @param kf_name Key name for the configuration file, such as @c "menu_new".
+ * @param label Label used in the preferences dialog keybindings tab. May contain
+ * underscores - these won't be displayed.
+ * @param menu_item Optional widget to set an accelerator for, or @c NULL.
+ * @param cb New-style callback to be called when activated, or @c NULL to use the group callback.
+ * @param pdata Plugin-specific data passed back to the callback.
+ * @param destroy_notify Function that is invoked to free the plugin data when not needed anymore.
+ * @return The keybinding - normally this is ignored.
+ *
+ * @since 1.25
+ * @see See plugin_set_key_group_full
+ **/
+GEANY_API_SYMBOL
+GeanyKeyBinding *keybindings_set_item_full(GeanyKeyGroup *group, gsize key_id,
+		guint key, GdkModifierType mod, const gchar *kf_name, const gchar *label,
+		GtkWidget *menu_item, GeanyKeyBindingFunc cb, gpointer pdata,
+		GDestroyNotify destroy_notify)
+{
+	GeanyKeyBinding *kb;
+
+	/* For now, this is intended for plugins only */
+	g_assert(group->plugin);
+
+	kb = keybindings_set_item(group, key_id, NULL, 0, 0, kf_name, label, menu_item);
```

Comment:
Am 18. Juni 2015 01:56:55 MESZ, schrieb Colomban Wendling <notifications@github.com>:
>> + *
>> + * @since 1.25
>> + * @see See plugin_set_key_group_full
>> + **/
>> +GEANY_API_SYMBOL
>> +GeanyKeyBinding *keybindings_set_item_full(GeanyKeyGroup *group,
>gsize key_id,
>> +		guint key, GdkModifierType mod, const gchar *kf_name, const gchar
>*label,
>> +		GtkWidget *menu_item, GeanyKeyBindingFunc cb, gpointer pdata,
>> +		GDestroyNotify destroy_notify)
>> +{
>> +	GeanyKeyBinding *kb;
>> +
>> +	/* For now, this is intended for plugins only */
>> +	g_assert(group->plugin);
>> +
>> +	kb = keybindings_set_item(group, key_id, NULL, 0, 0, kf_name,
>label, menu_item);
>
>hum, the function does have `key` and `mod` parameters, but they aren't
>propagated
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/geany/geany/pull/376/files#r32688415

Oops good catch, clearly a bug.